catroid/src/main/java/org/catrobat/catroid/transfers/RegistrationTask.java

```diff
@@ -134,12 +137,47 @@ private void showDialog(int messageId) {
 		if (message == null) {
 			new AlertDialog.Builder(context).setTitle(R.string.register_error).setMessage(messageId)
 					.setPositiveButton(R.string.ok, null).show();
+		} else if (message.equals(Constants.MISSING_PASSWORD)) {
+			showDialogWithMessage(R.string.register_error_missing_password);
+		} else if (message.equals(Constants.NOT_BLANK_USERNAME)) {
+			showDialogWithMessage(R.string.register_error_blank_username);
+		} else if (message.equals(Constants.NOT_BLANK_EMAIL)) {
+			showDialogWithMessage(R.string.register_error_blank_email);
+		} else if (message.equals(Constants.PASSWORD_LENGTH)) {
+			showDialogWithMessage(R.string.register_error_password_length);
+		} else if (message.equals(Constants.INVALID_COUNTRY)) {
+			showDialogWithMessage(R.string.register_error_invalid_country);
+		} else if (message.equals(Constants.USERNAME_EXISTS)) {
+			showDialogWithMessage(R.string.register_error_username_exists);
+		} else if (message.equals(Constants.INVALID_USERNAME_VALUE)) {
+			showDialogWithMessage(R.string.register_username_invalid_input_value);
+		} else if (message.equals(Constants.INVALID_EMAIL_ADDRESS)) {
+			showDialogWithMessage(R.string.register_invalid_email_address);
+		} else if (message.equals(Constants.EMAIL_ADDRESS_EXISTS)) {
+			new AlertDialog.Builder(context).setTitle(R.string.register_error).setMessage(R.string.register_email_exists_reset_password)
+					.setPositiveButton(R.string.ok, null)
+					.setNeutralButton(R.string.register_reset_password_btn, new DialogInterface.OnClickListener() {
+						@Override
+						public void onClick(DialogInterface dialog, int which) {
+							openResetPasswordWebPage();
+						}
+					}).show();
 		} else {
 			new AlertDialog.Builder(context).setTitle(R.string.register_error).setMessage(message)
 					.setPositiveButton(R.string.ok, null).show();
 		}
 	}
 
+	private void showDialogWithMessage(int message) {
+		new AlertDialog.Builder(context).setTitle(R.string.register_error).setMessage(message)
+				.setPositiveButton(R.string.ok, null).show();
+	}
+
+	private void openResetPasswordWebPage() {
+		Intent resetPassword = new Intent(Intent.ACTION_VIEW, Uri.parse(Constants.RESET_PASSWORD_URL));
+		context.startActivity(resetPassword);
```

Comment:
imo that's a great idea! I love it!

Can you please create a ticket for that so a ux member can review it and give some feedback/approve it.

Reply:
done! CAT-2733